cmd/simulator/metrics/metrics.go

```diff
@@ -0,0 +1,80 @@
+// (c) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// MonitorTPS periodically prints metrics related to transaction activity on
+// a given network.
+func MonitorTPS(ctx context.Context, client ethclient.Client) error {
+	blockNumber, err := client.BlockNumber(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get block number: %w", err)
+	}
+
+	block, err := client.BlockByNumber(ctx, new(big.Int).SetUint64(blockNumber))
+	if err != nil {
+		return fmt.Errorf("failed to get block at number %d: %w", blockNumber, err)
+	}
+
+	startTime := block.Time()
+	currentTime := startTime
+	totalTxs := 0
+	timeTxs := map[uint64]int{}
+	for ctx.Err() == nil {
+		newBlockNumber, err := client.BlockNumber(ctx)
+		if err != nil {
+			log.Printf("failed to get block number: %s", err)
+			time.Sleep(2 * time.Second)
+			continue
+		}
+
+		if blockNumber > newBlockNumber {
+			time.Sleep(2 * time.Second)
+			continue
+		}
```

Comment:
Should this be >=? I don't think we should ever regress backwards (assuming we are running with a normal config that only fetches accepted blocks).